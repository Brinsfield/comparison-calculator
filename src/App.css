body {
  background-color: #fbfbfb;
  color: #444;
  font-family: Lato, sans-serif;
}

#root {
  display: flex;
  justify-content: center;
}
.App {
  text-align: center;
  margin-top: 5em;
  /* background-color: #fff; */
  max-width: 1000px;
  padding: 3em;
  width: 700px;
}

.incompleted::before {
  content: '\f077';
}
