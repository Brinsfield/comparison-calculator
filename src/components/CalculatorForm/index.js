import React, { Component } from 'react';
class CalculatorForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      brand: '',
      nnn: 'Heartland Beef Mash',
      protein: 0,
      fat: 0,
      caloric: 0
    };

    this.handleChangeBrand = this.handleChangeBrand.bind(this);
    this.handleChangeNNN = this.handleChangeNNN.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state !== prevState) {
      if (
        this.state.brand &&
        this.state.protein &&
        this.state.fat &&
        this.state.caloric
      ) {
        this.props.completedForm(true);
        this.props.comparisonInfo(this.state);
      } else {
        this.props.completedForm(false);
      }
    }
  }

  handleChangeBrand(evt) {
    this.setState({
      brand: evt.target.value
    });
  }
  handleChangeNNN(evt) {
    this.setState({
      nnn: evt.target.value
    });
  }

  handleInputChange(evt) {
    this.setState({
      [evt.target.name]: +evt.target.value
    });
  }

  render() {
    let infoText = this.state.brand
      ? `${this.state.brand}'s nutrient info:`
      : `Dog food brand's nutrient info:`;

    return (
      <form>
        <label>Name of dog food brand</label>
        <select value={this.state.brand} onChange={this.handleChangeBrand}>
          <option value="">-Select a Brand-</option>
          <option value="Bonio">Bonio</option>
          <option value="Milk-Bone">Milk-Bone</option>
          <option value="Friskies">Friskies</option>
        </select>

        <label>Compare with NomNomNow</label>
        <select value={this.state.nnn} onChange={this.handleChangeNNN}>
          <option value="Porkalicious Potluck">Porkalicious Potluck</option>
          <option defaultValue="Heartland Beef Mash">
            Heartland Beef Mash
          </option>
          <option value="Chicken Chow-Wow">Chicken Chow-Wow</option>
          <option value="Tasty Turkey Fare">Tasty Turkey Fare</option>
        </select>

        <div>{infoText}</div>

        <label>
          % protein (min.)
          <input
            type="number"
            name="protein"
            onChange={this.handleInputChange}
          />
        </label>

        <label>
          % fat (min.)
          <input type="number" name="fat" onChange={this.handleInputChange} />
        </label>

        <label>
          Caloric density (kcal/kg)
          <input
            type="number"
            name="caloric"
            onChange={this.handleInputChange}
          />
        </label>
      </form>
    );
  }
}

export default CalculatorForm;
