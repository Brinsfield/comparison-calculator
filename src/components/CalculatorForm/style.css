.calculator-form {
  text-align: left;
}

.food-brand,
.inputs {
  display: flex;
  padding-bottom: 10px;
}

.other-food-brands,
.nnn-food-brand {
  width: 100%;
  padding-bottom: 10px;
}

.other-food-brands {
  padding-right: 10px;
}

.label {
  display: block;
  font-weight: 400;
  padding-bottom: 5px;
}

select {
  background: url('../../assets/icon_chevron.png') no-repeat 96% center;
  background-color: #fff;
  width: 100%;
  background-size: 10px;
  border: 1px solid #eee;
  border-radius: 2px;

  font-size: 1em;
  padding: 10px 30px 10px 10px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.padding-bottom-small {
  padding-bottom: 10px;
}

.padding-right-small {
  padding-right: 10px;
}

.inputs div {
  width: 100%;
}

input {
  background-color: #fff;
  border: 1px solid #eee;

  font-size: 1em;
  padding: 10px 0;
  width: 100%;
}
