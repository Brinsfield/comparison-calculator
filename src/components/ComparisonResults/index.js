import React, { Component } from 'react';
import logo from '../../assets/logo_utensils.svg';

class ComparisonResults extends Component {
  state = {
    protein: 0,
    fat: 0,
    carbohydrat: 0
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    const formData = nextProps.formData;

    let p = (formData.protein + 1.5) / (formData.caloric / 10000);
    let f = (formData.fat + 1) / (formData.caloric / 10000);
    let c = (1000 - p * 3.5 - f * 8.5) / 3.5;
    c = c < 0 ? 0 : c;

    if (
      p !== prevState.protein ||
      f !== prevState.fat ||
      c !== prevState.carbohydrat
    ) {
      return {
        protein: p,
        fat: f,
        carbohydrat: c
      };
    }

    return null;
  }

  nomnomnowData(nnn) {
    const HBM = { protein: 80, fat: 62, carbohydrate: 55 };
    const PP = { protein: 78, fat: 63, carbohydrate: 56 };
    const CCW = { protein: 115, fat: 58, carbohydrate: 31 };
    const TTF = { protein: 96, fat: 46, carbohydrate: 78 };

    switch (nnn) {
      case 'Porkalicious Potluck':
        return PP;
      case 'Chicken Chow-Wow':
        return CCW;
      case 'Tasty Turkey Fare':
        return TTF;
      default:
        return HBM;
    }
  }

  render() {
    const { brand, nnn } = this.props.formData;
    const nnnFoodData = this.nomnomnowData(nnn);

    let otherBrand = (
      <div>
        <h3>{brand}</h3>
        <label>Protein</label>
        <h1>{this.state.protein}</h1>
        <label>Fat</label>
        <h1>{this.state.fat}</h1>
        <label>Carbohydrate</label>
        <h1>{this.state.carbohydrat}</h1>
      </div>
    );

    let nomnomnow = (
      <div>
        <img src={logo} alt="logo" />
        <h3>{nnn}</h3>
        <label>Protein *</label>
        <h1>{nnnFoodData.protein}</h1>
        <label>Fat</label>
        <h1>{nnnFoodData.fat}</h1>
        <label>Carbohydrate</label>
        <h1>{nnnFoodData.carbohydrate}</h1>
      </div>
    );

    return (
      <div>
        <div>{otherBrand}</div>
        <div>{nomnomnow}</div>
        <h5>
          Learn about the role of protein, fat, and carbs in a dog's diet.
        </h5>
        <h5>
          * All units above are given in grams per 1000 calories (g/kcal).
        </h5>
      </div>
    );
  }
}

export default ComparisonResults;
