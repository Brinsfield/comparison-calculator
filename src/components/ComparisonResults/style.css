.result-container {
  background-color: rgba(255, 255, 255, 0.8);
  padding: 20px;
  color: #818181;
}

.small-logo-icon {
  display: block;
  height: 40px;
  width: auto;
  margin-bottom: 10px;
}

.results {
  display: flex;
}

.results > div {
  padding: 20px 0 0 20px;
  width: 100%;
}
.results > div:first-child {
  border-right: 2px solid #e0e0e0;
  padding: 70px 20px 0 0;
  text-align: right;
}

.results > div:nth-child(2) {
  text-align: left;
}

h3.brand {
  font-weight: 300;
  line-height: 1.1em;
  height: 60px;
  margin: 0;
  font-size: 22px;
  color: #444444;
}

h1 {
  font-weight: 300;
  line-height: 1.1em;
  margin-bottom: 20px;
  font-size: 24px;
}

.color-orange {
  color: #f06324;
}
.color-teal {
  color: #23ac9b;
}

.color-yellow {
  color: #e2d16e;
}

a {
  color: #f06324;
  text-decoration: none;
  font-weight: 400;
}

.text-smallest {
  font-size: 13px;
}

.padding-bottom {
  padding-bottom: 20px;
}
