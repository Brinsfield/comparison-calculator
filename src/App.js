import React, { Component } from 'react';
import CalculatorForm from './components/CalculatorForm';
import ComparisonResults from './components/ComparisonResults';
import './App.css';

class App extends Component {
  state = {
    isFilled: false,
    comparisonInfo: null
  };

  completedForm = isCompleted => {
    this.setState({
      isFilled: isCompleted
    });
  };

  getComparisonInfo = info => {
    this.setState({
      comparisonInfo: info
    });
  };

  render() {
    let confirmedCompletedMsg = !this.state.isFilled ? (
      <h4>
        <div>
          <i className="fas fa-chevron-up" />
        </div>
        Fill out the form to get a nutrient comparison
      </h4>
    ) : (
      <h4>
        See the the nutrient comparison below
        <div>
          <i className="fas fa-chevron-down" />
        </div>
      </h4>
    );

    let comparisonResults = this.state.isFilled ? (
      <ComparisonResults formData={this.state.comparisonInfo} />
    ) : null;

    return (
      <div className="App">
        <CalculatorForm
          completedForm={this.completedForm}
          comparisonInfo={this.getComparisonInfo}
        />
        {confirmedCompletedMsg}
        {comparisonResults}
      </div>
    );
  }
}

export default App;
