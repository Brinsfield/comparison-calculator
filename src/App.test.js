import React from 'react';
import { shallow, mount } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import App from './App';
import CalculatorForm from './components/CalculatorForm';
import ComparisonResults from './components/ComparisonResults';

describe('<App />', () => {
  let wrapper;
  it('renders', () => {
    // smoke test!!!
    wrapper = shallow(<App />);
  });
  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});

describe('full rendering and tesing <Calculator> funcitonality', () => {
  let wrapper;
  it('full rendering', () => {
    wrapper = mount(<App />);
    expect(wrapper.find(CalculatorForm).length).toEqual(1);

    wrapper
      .find('.other-food-brands')
      .find('select')
      .find('option')
      .at(1)
      .simulate('change', {
        target: { value: 'Bonio' }
      });

    wrapper
      .find(CalculatorForm)
      .find('.inputs')
      .find('input')
      .at(0)
      .simulate('change', {
        target: { name: 'protein', value: 1 }
      });

    wrapper
      .find(CalculatorForm)
      .find('.inputs')
      .find('input')
      .at(1)
      .simulate('change', {
        target: { name: 'fat', value: 2 }
      });

    wrapper
      .find(CalculatorForm)
      .find('.inputs')
      .find('input')
      .at(2)
      .simulate('change', {
        target: { name: 'caloric', value: 3 }
      });

    let obj = {
      brand: 'Bonio',
      caloric: 3,
      fat: 2,
      nnn: 'Heartland Beef Mash',
      protein: 1
    };
    expect(wrapper.find(CalculatorForm).instance().state).toEqual(obj);
    expect(wrapper.find(ComparisonResults).length).toEqual(1);
  });
});
