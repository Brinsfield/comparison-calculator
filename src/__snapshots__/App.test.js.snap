// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <CalculatorForm
    comparisonInfo={[Function]}
    completedForm={[Function]}
  />
  <h4>
    <div>
      <i
        className="fas fa-chevron-up"
      />
    </div>
    Fill out the form to get a nutrient comparison
  </h4>
</div>
`;
